graviti/openapi/dataset.py

```diff
@@ -0,0 +1,239 @@
+                   "default_branch": "main",
+                   "commit_id": "a0d4065872f245e4ad1d0d1186e3d397",
+                   "cover_url": "https://tutu.s3.cn-northwest-1.amazonaws.com.cn/",
+                   "created_at": "2021-03-03T18:58:10Z",
+                   "updated_at": "2021-03-03T18:58:10Z",
+                   "owner": "czhual",
+                   "is_public": false,
+                   "config_name": "exampleConfigName"
+               }
+            ],
+           "offset": 0,
+           "record_size": 1,
+           "total_count": 1
+        }
+
+
+    """
+    url = urljoin(url, f"{URL_PATH_PREFIX}/datasets")
+    params: Dict[str, Any] = {"offset": offset, "limit": limit}
+
+    return open_api_do("GET", access_key, url, params=params).json()  # type: ignore[no-any-return]
+
+
+def update_dataset(
+    access_key: str,
+    url: str,
+    dataset_name: str,
+    *,
+    name: Optional[str] = None,
+    alias: Optional[str] = None,
+    is_public: Optional[bool] = None,
+    default_branch: Optional[str] = None,
+) -> None:
+    """Execute the OpenAPI `PATCH /v1/datasets/{dataset_name}`.
+
+    Arguments:
+        access_key: User's access key.
+        url: The URL of the graviti website.
+        dataset_name: Name of the dataset, unique for a user.
+        name: New name of the dataset, unique for a user.
+        alias: New alias of the dataset.
+        is_public: Whether the dataset is public.
+        default_branch: User's chosen branch.
+
+    Examples:
+        Rename dataset:
+
+        >>> update_dataset(
+        ...     "ACCESSKEY-********",
+        ...     "https://graviti.com/",
+        ...     "OxfordIIITPet",
+        ...     "Oxford-IIIT Pet"
+        ... )
+
+        Update the information about alias, default branch and whether the dataset is public:
+        >>> update_dataset(
+        ...     "ACCESSKEY-********",
+        ...     "https://graviti.com/",
+        ...     "OxfordIIITPet",
+        ...     alias="Oxford-IIIT Pet",
+        ...     is_public=True,
+        ...     default_branch="main",
+        ... )
+
+
+    """
+    url = urljoin(url, f"{URL_PATH_PREFIX}/datasets/{dataset_name}")
+    patch_data: Dict[str, Any] = {}
+
+    if name is not None:
+        patch_data["name"] = name
+
+    if alias is not None:
+        patch_data["alias"] = alias
+
+    if is_public is not None:
+        patch_data["is_public"] = is_public
+
+    if default_branch is not None:
+        patch_data["default_branch"] = default_branch
+
+    open_api_do("PATCH", access_key, url, json=patch_data)
+
+
+def delete_dataset(
+    access_key: str,
+    url: str,
+    dataset_name: str,
+) -> None:
+    """Execute the OpenAPI `DELETE /v1/datasets/{dataset_name}`.
+
+    Arguments:
+        access_key: User's access key.
+        url: The URL of the graviti website.
+        dataset_name: Name of the dataset, unique for a user.
+
+    Examples:
+        Delete dataset:
+
```

Comment:
There is only one example, we do not add information unless it needs to be emphasized.
Other areas need fixing as well.